docs/public/specifications/tests/DNSSEC-TP/dnssec03.md

```diff
@@ -1,73 +1,316 @@
+the majority of the records are insecure delegations, opt-out MAY be used"
+([RFC 9276][RFC 9276#section-3.1], section 3.1). This means that unless the zone
+is a TLD or a TLD like domain found in the [Public Suffix List] it should
+not have the opt-out bit set.
+
+**Iterations:** For a name server an increased number of NSEC3 iterations have a
+negative impact on performance. The recommendation is to have 0 iterations. "If
+NSEC3 must be used, then an iterations count of 0 MUST be used to alleviate
+computational burdens" ([RFC 9276][RFC 9276#section-3.1], section 3.1).
+
+**Salt:** The salt parameter has been seen as a security feature but
+[RFC 9276][RFC 9276#section-3.1], section 3.1, states that zones "SHOULD NOT use
+a salt by indicating a zero-length salt value instead". The justification for
+the recommendation is found in [RFC 9276][RFC 9276#section-2.4], section 2.4.
+
+
+## Scope
+
+This test case will just ignore non-responsive name servers or name servers not
+giving a correct DNS response for an authoritative name server (covered by
+[Connectivity01]).
+
+This test case is only relevant if the zone has been DNSSEC signed.
+
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+* "Public Suffix List Data" - The list or a copy of the list found at
+  [Public Suffix List data].
+
+
+## Summary
+
+* If no DNSKEY records are found, no further investigation will be done.
 
-### Objective
+Message Tag outputted              | Level   | Arguments  | Message ID for message tag
+:----------------------------------|:--------|:-----------|:--------------------------------------------
+DS03_ERR_MULT_NSEC3                | ERROR   | ns_list | Multiple NSEC3 records when one is expected. Fetched from name servers "{ns_list}".
+DS03_ILLEGAL_HASH_ALGO             | ERRRO   | ns_list, algo_num | The following servers respond with an illegal hash algorithm for NSEC3 ({algo_num}). Fetched from name servers "{ns_list}".
+DS03_ILLEGAL_ITERATION_VALUE       | ERROR   | ns_list, int | The following servers respond with the NSEC3 iteration value {int} (not recommended). Fetched from name servers "{ns_list}".
+DS03_ILLEGAL_SALT_LENGTH           | WARNING | ns_list, int | The following servers respond with a non-empty salt in NSEC3 ({int} octets). Fetched from name servers "{ns_list}".
+DS03_INCONSISTENT_HASH_ALGO        | ERROR   |         | Inconsistent hash algorithms in NSEC3 in responses for the child zone from different name servers.
+DS03_INCONSISTENT_ITERATION        | ERROR   |         | Inconsistent NSEC3 iteration values in responses for the child zone from different name servers.
+DS03_INCONSISTENT_NSEC3_FLAGS      | ERROR   |         | Inconsistent NSEC3 flags in responses for the child zone from different name servers.
+DS03_INCONSISTENT_SALT_LENGTH      | ERROR   |         | Inconsistent salt length in NSEC3 in responses for the child zone from different name servers.
```

Comment:
```suggestion
DS03_INCONSISTENT_SALT_LENGTH      | ERROR   |         | Inconsistent salt lengths in NSEC3 in responses for the child zone from different name servers.
```

Reply:
I would say singular since there could only be one length.

Reply:
I don’t think so. We’re talking about several values, always two or more, that do not agree between each other. In the singular form, it implies that it’s self-contradictory (such as in “an inconsistent story”), which for a numeric value, makes less sense. Besides, we have inconsistent messages: the previous messages use “hash algorithms” and “iteration values” in their plural forms.

Reply:
I checked with a native English speaker with two of the messages, and he suggests singular.